### PR TITLE
Fix warnings found by analysis tools

### PR DESCRIPTION
**Pull request by bwester**
Simplifies the source code formatting via "gofmt" and fixes many questionable practices flagged by "go vet".

### CODE REVIEW

File: pkg/kp/consulutil/fake_kv.go [hunk 1]
```diff
@@ -31,34 +31,34 @@ func NewFakeClient() *FakeConsulClient {
 	}
 }
 
-func NewKVWithEntries(entries map[string]*api.KVPair) FakeKV {
+func NewKVWithEntries(entries map[string]*api.KVPair) *FakeKV {
```
Comment (mpuncel) [inline pkg/kp/consulutil/fake_kv.go]: i'm curious why this change was necessary?

File: pkg/util/stream/publisher_test.go [hunk 2]
```diff
@@ -59,7 +59,7 @@ RecvLoop:
 	for _, r := range received {
 		for i, s := range sent {
 			if r == s {
-				sent = sent[i+1 : len(sent)]
+				sent = sent[i+1:]
```
Comment (mpuncel) [inline pkg/util/stream/publisher_test.go]: interesting thought Go didn't let you do this

File: pkg/kp/consulutil/fake_kv.go (hunk 1 above)
Comment (bwester) [inline pkg/kp/consulutil/fake_kv.go]: `FakeKV` had a `sync.Mutex` in it, and mutexes can't be copied.

**Comment by JackDanger** [comment]
This is great, @bwester - now when an editor is configured to run `go vet` it'll complain less. Thanks!
